packages/lodestar/src/network/peers/datastore.ts

```diff
@@ -0,0 +1,187 @@
+import {BaseDatastore} from "datastore-core";
+import LevelDatastore from "datastore-level";
+import {Key, KeyQuery, Query, Options, Pair} from "interface-datastore";
+
+type MemoryItem = {
+  lastAccessedMs: number;
+  data: Uint8Array;
+};
+
+/**
+ * Before libp2p 0.35, peerstore stays in memory and periodically write to db after n dirty items
+ * This has a memory issue because all peer data stays in memory and loaded at startup time
```

Comment:
Have we ever documented a memory issue specifically due to the peerstore? The information for the peerstore doesn't have to be kept around forever